### DISCUSSION TITLE
Switiching from '!!' to '===' for 'process.env.D365_UI_TEST_HEADLESS'

### DISCUSSION THREAD

**Discussion by julianYaman** [XRM-OSS/D365-UI-Test#36] [Ideas]
Hello,
I thought to move my suggestion to the "Discussions" section since it is not really an issue and IMO does not need a fix with a PR.

Today, I was working with environment variables to launch the browser in the `beforeAll` statement and had problems with using the approach from the README
```js
const response = await xrmTest.launch("chromium", {
    headless: !!process.env.D365_UI_TEST_HEADLESS,
    args: [
      "--disable-setuid-sandbox",
      "--disable-infobars",
      "--start-fullscreen",
      "--window-position=0,0",
      "--window-size=1920,1080",
      '--user-agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36"',
    ],
});
```

I use `dotenv` and a local `.env` file to set the value of `D365_UI_TEST_HEADLESS` by writing `D365_UI_TEST_HEADLESS=false`.

My workaround was to use this approach:
```js
const response = await xrmTest.launch("chromium", {
    headless: process.env.D365_UI_TEST_HEADLESS === "true",
    args: [
      "--disable-setuid-sandbox",
      "--disable-infobars",
      "--start-fullscreen",
      "--window-position=0,0",
      "--window-size=1920,1080",
      '--user-agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36"',
    ],
});
```

I decided to use this approach after reading [this on StackOverflow](https://stackoverflow.com/questions/263965/how-can-i-convert-a-string-to-boolean-in-javascript).

Would it be better to switch from `!!` to `===`?
It would definitely mean that developers are only allowed to set 'true' explicitly, otherwise they would get `false` as the value for `D365_UI_TEST_HEADLESS`.

What do you think about this?

**Comment by imartinflores** [comment]
hey @julianYaman ! I am not very experienced in JS, but if that's suggested in StackOverflow and works better, then it should be ok to update it!

**Reply by julianYaman** [reply]
Also thought this. Seems that using `!!` may not be recommended, because of the explanation in the StackOverflow post:

> Any string which isn't the empty string will evaluate to true by using them. 

Hope this is ok for @DigitalFlow to change it too.

**Comment by DigitalFlow** [comment]
Hi @julianYaman and @imartinflores,

you're totally right, the `===`  approach is better.
There is an open task for migrating to `dotenv` as well, as the current parsing is not optimal either.

We'll implement these two changes in one commit 👍 

Kind regards,
Florian

**Reply by julianYaman** [reply]
Great to hear that! 😄
